### PR TITLE
Add "uvToRenderTargetUV" to public shading API.

### LINKED ISSUE #1626: Vulkan and Metal: lucy_bloom is upside down
Other samples are fine, this is probably related to the new RenderTarget API.

### PR DESCRIPTION
**Pull request by prideout**
Fixes #1626.

### CODE REVIEW

File: shaders/src/getters.fs [hunk 1]
```diff
@@ -61,6 +61,25 @@ float getNdotV() {
...
+ * In Vulkan and Metal, texture coords are Y-down. In OpenGL, texture coords are Y-up.
+ *
+ * When sampling from non-render targets (i.e. uploaded textures) this distinction does not matter
+ * because OpenGL has a second piece of backwardness, which is that the first row of texels
+ * in glTexImage2D is interpreted as the bottom row.
+ *
+ * To protect users from these differences, we recommend that materials in the SURFACE domain
+ * leverage this wrapper function when sampling from offscreen render targets.
+ */
+vec4 sampleRenderTarget2D(sampler2D samp, vec2 uv) {
```
Comment (bejado) [inline shaders/src/getters.fs]: Maybe make note that these are normalized uv coordinates? In the future we might want to add a `texelFetchRenderTarget` method too that uses texel coordinates.

Comment (prideout) [reply shaders/src/getters.fs]: Done.

File: docs/Materials.md.html [hunk 2]
```diff
@@ -1610,6 +1610,7 @@
 **getWorldGeometricNormalVector()** | float3   |  Normalized normal in world space, before bump mapping (can be used before `prepareMaterial()`)
 **getWorldReflectedVector()**       | float3   |  Reflection of the view vector about the normal (must be used after `prepareMaterial()`)
 **getNdotV()**                      | float    |  The result of `dot(normal, view)`, always strictly greater than 0 (must be used after `prepareMaterial()`)
+**sampleRenderTarget2D(sampler2D, vec2)** | float4 | Similar to the GLSL built-in `texture` but for offscreen render targets
```
Comment (romainguy) [inline docs/Materials.md.html]: Could we match the `texture()` name instead? Maybe `renderTarget()`? Also can you not put it in the middle of the get methods in this list?

File: docs/Materials.md.html [hunk 3]
```diff
@@ -1622,6 +1623,12 @@
     To obtain API-level world space coordinates, custom materials should add `getWorldOffset()` to
     `getWorldPosition()` (et al).
 
+!!! TIP: sampling from render targets
+    When writing materials for the surface domain, please use `sampleRenderTarget2D` instead of
```
Comment (romainguy) [inline docs/Materials.md.html]: Did you mean `post process` instead of `surface`?

Comment (prideout) [reply docs/Materials.md.html]: I actually mean `surface` here.  In a post-process shader, the uv we provide is already flipped.  In the surface domain, we do not provide uv's so we provide this helper instead.

File: docs/Materials.md.html (hunk 2 above)
Comment (romainguy) [reply docs/Materials.md.html]: Ah and you should also align the table separators

File: docs/Materials.md.html (hunk 3 above)
Comment (romainguy) [reply docs/Materials.md.html]: I see, that's a little weird. I wonder if instead we shouldn't provide a method to flip the UVs: `uvToRenderTargetUV()` and just use `texture()` as usual.

Comment (prideout) [reply docs/Materials.md.html]: I'm fine with that idea, it will be more verbose but it will also be more flexible.

**Comment by prideout** [comment]
Ok -- I've amended the commit to incorporate feedback from @romainguy and the bloom demo still looks ok with Metal & Vulkan.

**Pull request merged**